controller/nodelabel_controller.go

```diff
@@ -405,11 +377,28 @@ func (r *ReconcileNodeLabel) applyTagsToNodes(namespacedName types.NamespacedNam
 		}
 	}
 
-	if !changed { // to avoid unnecessary patching
+	// delete labels if tag has been deleted
+	// if conflict policy is node precedence (which it will most likely not be), then don't delete tags if they exist on node
+	if configOptions.LabelPrefix != "" && (configOptions.ConflictPolicy == ARMPrecedence || configOptions.ConflictPolicy == Ignore) {
```

Comment:
consider encapsulating the if clause in a func with a meaningful name (your comment says if conflict policy is node precedence).
if possible, extract the content of the if in a separate func too. it helps for readability